src/raw_rwlock.rs

```diff
@@ -13,19 +13,53 @@ use stable::{AtomicUsize, Ordering};
 use std::time::{Duration, Instant};
 use parking_lot_core::{self, ParkResult, UnparkResult, SpinWait, ParkToken, FilterOp};
 use elision::{have_elision, AtomicElisionExt};
-use util::UncheckedOptionExt;
 use raw_mutex::{TOKEN_NORMAL, TOKEN_HANDOFF};
 use deadlock;
 
+const USABLE_BITS_MASK: usize = {
+    #[cfg(feature = "nightly")] {
+        const TOTAL_BITS: usize = ::std::mem::size_of::<usize>() * 8;
+        // specifies the number of usable bits, useful to test the
+        // implementation with fewer bits (the current implementation
+        // requires this to be at least 4)
+        const USABLE_BITS: usize = TOTAL_BITS;
+        (!0) >> (TOTAL_BITS - USABLE_BITS)
+    }
+    #[cfg(not(feature = "nightly"))] {
+        !0
+    }
+};
```

Comment:
What is the purpose of this constant? It seems that it always has the value `!0`.

Reply:
I personally need to include this in a more complicated data structure (tagged pointers and the like...), so I wanted to make sure the implementation works and doesn't panic when using only a few bits for the state.

I added this so that it was easy for me to switch between testing with just a few bits and when using the full `usize`. In particular, everything should work perfectly fine with a value of `0b1111`, just with some different performance characteristics.